src/core/evaluator.js

```diff
@@ -1997,23 +1998,27 @@ class PartialEvaluator {
       width: 0,
       height: 0,
       vertical: false,
+      lastCharSize: 0,
       lastAdvanceWidth: 0,
       lastAdvanceHeight: 0,
       textAdvanceScale: 0,
       spaceWidth: 0,
       fakeSpaceMin: Infinity,
-      fakeMultiSpaceMin: Infinity,
-      fakeMultiSpaceMax: -0,
-      textRunBreakAllowed: false,
       transform: null,
       fontName: null,
+      isSpace: false,
+      hasEOL: false,
     };
-    var SPACE_FACTOR = 0.3;
-    var MULTI_SPACE_FACTOR = 1.5;
-    var MULTI_SPACE_FACTOR_MAX = 4;
+
+    // Used in addFakeSpaces
+    // A white = whiteSpace +/- (100*SPACE_ERROR)% is a white space.
+    const SPACE_ERROR = 0.1;
```

Comment:
`0.2` doesn't change the order of magnitude.
Anyway I split the `SPACE_ERROR` value into two: a lower and an upper bounds.
Setting `SPACE_IN_FLOW_MIN_FACTOR` to `0.25` leads to `2901` spans for all the document which is almost what we've right now.
As far as I understood, "enhanced" text-selection is mainly a chrome feature, correct ? if yes then we can tune the constant above to avoid perf regression when enhanced text-selection is enabled.